test/plugins/test_replaygain.py

```diff
@@ -51,6 +52,36 @@ def reset_replaygain(item):
     item.store()
 
 
+class ReplayGainTestCase(ImportTestCase):
+    db_on_disk = True
+    backend: ClassVar[str]
+
+    def setUp(self):
+        # Implemented by Mixins, see above. This may decide to skip the test.
+        self.test_backend()
+
+        super().setUp()
+        self.config["replaygain"]["backend"] = self.backend
+
+        try:
+            self.load_plugins("replaygain")
+        except Exception:
+            self.tearDown()
```

Comment:
Why not use `PluginTestCase` here?  And why do you have to manually call `tearDown()` if an exception occurs? (won't `pytest` do that for you automatically?)

Reply:
You will see there are a more test modules which run `load_plugins` and `unload_plugins` manually and they all have something in common: they set up plugin configuration before loading it.

Some of them test various configurations, like `test/plugin/test_filtefilter.py`, where the plugin is loaded midway through the test.

`PluginMixin / PluginTestCase` loads the plugin at the end of the setup, before the test methods runs, therefore plugin configuration defined in the test method simply has no effect. This mixin is only useful for those plugin tests that simply need the plugin to be loaded (majority of them). Otherwise, as you see here, the plugin is loaded manually.

I think it may be a good idea to define a method `PluginMixin.reload_plugin` which does something like

```python
class PluginMixin:
    ...
    def reload_plugin(self, config: dict):
        self.unload_plugins()
        beets.config[self.plugin].update(config)
        self.load_plugins(self.plugin)
```

Reply:
Right, thank you for explaining!  But would a `test_plugin_load()` method (set config; load plugin; unload plugin) be more appropriate, if tests call `reload_plugin()` to try loading a plugin temporarily?

Reply:
@bal-e I'm not sure whether I understood what you meant correctly. Maybe something like

```python
from contextlib import contextmanager

...

class PluginMixin:
    @contextmanager
    def configure_plugin(self, config):
        beets.config[self.plugin].update(config)
        self.load_plugins(self.plugin)

        yield

        self.unload_plugins()

class PluginTest(PluginMixin, ...):
    def test_plugin_config(self):
        with self.configure_plugin({"option": "value"}):
            self.assertEqual(beets.config[self.plugin]["option"].get(), "value")
```

This would clean up the configuration on test exit.

Reply:
I'll point out that this context manager approach is very, very similar to the pytest approach with fixtures. If we do that, it'll be less of a change when converting the remainder of the files.

Reply:
@Serene-Arc completely agree. I have done this in 28ae414.